src/zenml/zen_server/routers/pipeline_deployments_endpoints.py

```diff
@@ -179,33 +181,65 @@ def delete_deployment(
     )
 
 
-if server_config().workload_manager_enabled:
+@router.get(
+    "/{deployment_id}/logs",
+    responses={
+        401: error_response,
+        404: error_response,
+        422: error_response,
+    },
+)
+@handle_exceptions
+def deployment_logs(
+    deployment_id: UUID,
+    offset: int = 0,
+    length: int = 1024 * 1024 * 16,  # Default to 16MiB of data
+    _: AuthContext = Security(authorize),
+) -> str:
+    """Get deployment logs.
+
+    Args:
+        deployment_id: ID of the deployment.
+        offset: The offset from which to start reading.
+        length: The amount of bytes that should be read.
+
+    Returns:
+        The deployment logs.
+    """
+    store = zen_store()
 
-    @router.get(
-        "/{deployment_id}/logs",
-        responses={
-            401: error_response,
-            404: error_response,
-            422: error_response,
-        },
+    deployment = verify_permissions_and_get_entity(
+        id=deployment_id,
+        get_method=store.get_deployment,
+        hydrate=True,
     )
-    @handle_exceptions
-    def deployment_logs(
-        deployment_id: UUID,
-        _: AuthContext = Security(authorize),
-    ) -> str:
-        """Get deployment logs.
-
-        Args:
-            deployment_id: ID of the deployment.
-
-        Returns:
-            The deployment logs.
-        """
-        deployment = verify_permissions_and_get_entity(
-            id=deployment_id,
-            get_method=zen_store().get_deployment,
-            hydrate=True,
-        )
 
+    if deployment.template_id and server_config().workload_manager_enabled:
         return workload_manager().get_logs(workload_id=deployment.id)
+
+    # Get the last pipeline run for this deployment
+    pipeline_runs = store.list_runs(
```

Comment:
This seems very dubious to me tbh. Shouldn't this simply be an endpoint that gets the logs for a run?

Reply:
I created another endpoint for that purpose. I'll also keep these changes in until the OSS dashboard is updated and we can remove this endpoint.